### DISCUSSION TITLE
wait until expected URL is loaded

### DISCUSSION THREAD

**Discussion by poojapatel14710** [seleniumbase/SeleniumBase#1515] [Q&A] [answered]
Hello sir!
 
 I am using the below code to my selenium web driver to wait until the expected URL is loaded, but I don't have any idea how to use that with the selenium base. could you please suggest to me a solution to this problem?
```
 
        wait = WebDriverWait(self.driver, 20)
        wait.until(lambda driver: driver.current_url == "https://www.google.com")
```

**Comment by mdmintz** [comment, marked as answer]
Hi @poojapatel14710 

You can do something like this:

```python
from selenium.webdriver.support.ui import WebDriverWait
from seleniumbase import BaseCase

class MyTestClass(BaseCase):
    def test_base(self):
        self.open("https://www.google.com/")
        wait = WebDriverWait(self.driver, 4)
        wait.until(lambda driver: driver.current_url == "https://www.google.com/")
```

And that will succeed assuming you're not redirected to a different URL. In the specific example above, the `WebDriverWait` is unnecessary, because SeleniumBase automatically waits for a page to finish loading before resuming automation.

**Reply by poojapatel14710** [reply]
Actually what I am doing is, I am clicking on the back icon of page and then need to check, clicking on the back icon successfully go back to the expected URL

```
from selenium.webdriver.support.wait import WebDriverWait
from .base_case import BaseTestCase
from .page_objects import Viber

class ViberNewChannelTest(BaseTestCase):

    def test_02_back_icon(self):
        self.open("https://......./channels/vbr/")  # using cookies to login and directly land on this page
        self.click(Viber.back_icon)
        wait = WebDriverWait(self.driver, 20)
        wait.until(lambda driver: driver.current_url == "https://........./channels/add")
```

**Reply by mdmintz** [reply]
Did that solve your issue?

**Reply by poojapatel14710** [reply]
Sir, The solution you gave me is the same as my code. but what I ask you, Is that the correct way in selenium base to verify the actual URL to the expected URL.? or is there any other way?

**Reply by mdmintz** [reply]
There's no specific method for asserting the URL, but there is a specific method for asserting the page title:
``self.assert_title(title)``. If asserting the title is not good enough, or asserting a specific element on the page, then your way should be good enough.

**Reply by poojapatel14710** [reply]
Okay so, I am in the right direction. Thank you so much Sir for your help.
